troposphere/__init__.py

```diff
@@ -162,10 +162,43 @@ def validate(self):
         pass
 
     @classmethod
-    def from_dict(cls, title, dict):
-        obj = cls(title)
-        obj.properties.update(dict)
-        return obj
+    def _from_dict(cls, title=None, **kwargs):
+        props = {}
+        for prop_name, value in kwargs.items():
+            try:
+                prop_attrs = cls.props[prop_name]
```

Comment:
oh you're right, this is the `cls` object. i was thinking of the function you had before that took any class